metadata-ingestion/src/datahub/configuration/common.py

```diff
@@ -278,7 +278,7 @@ class KeyValuePattern(ConfigModel):
     For example, you can use it to map a table name to a list of tags to apply to it.
     """
 
-    rules: Dict[str, List[str]] = {".*": []}
+    rules: Dict[str, Union[str, List[str]]] = {".*": []}
```

Comment:
this type should not change. instead, please add a pydantic validator that will convert a single value into a list

then the dataproduct transformer should throw a ConfigurationError if more than one value is specified